bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -198,7 +198,7 @@ public void loadScript(String engineIdentifier, InputStreamReader scriptData,
                     logger.trace("ScriptEngine does not support Invocable interface");
                 }
             } catch (Exception ex) {
-                logger.error("Error during evaluation of script '{}': {}", engineIdentifier, ex.getMessage());
+                logger.error("Error during evaluation of script '{}'", engineIdentifier, ex);
```

Comment:
on the JRuby side, this will always be a ScriptException, and it will always have a cause, and the cause will only have the stack trace from the Ruby code. So it'd be most useful to automatically unwrap one level down. But I'm not familiar with other script engines, and didn't want to lose information, so I just left it exposing the main exception.

Reply:
At JS Scripting, this will also be very useful for debugging.
We have a new issue, that relates to this (https://github.com/openhab/openhab-js/issues/176).

I haven’t tested how the stack trace looks for JS Scripting here, but in our case, the exception is thrown because the user passes a parameter of wrong type to a Java method.
I‘ll try to have check the stack trace later.

Reply:
FYI: Wrapping down wouldn’t work here, because for example in JS Scripting we can also have a `IllegalArgumentException` in the case described above.

Reply:
> FYI: Wrapping down wouldn’t work here, because for example in JS Scripting we can also have a `IllegalArgumentException` in the case described above.

In the jrubyscripting addon there are a couple of places that the engine is called directly to set things up, and exceptions are caught there. I _did_ do unwrapping of those, but did it in a safe manner in case there _isn't_ a cause:

https://github.com/openhab/openhab-addons/commit/4739f126a2172d43034bbe57e450817d46f6667d#diff-e433203688ef4765215fe6f7efe301f64fe4a0365cc635a07c3a91adc18d6f0fR296

Reply:
I‘ve just checked the JS Scripting code, there is already some code for the logging in place, I‘ll see if I can improve it.